src/luv.c

```diff
@@ -18,7 +18,13 @@
 #if (LUA_VERSION_NUM != 503)
 #include "c-api/compat-5.3.h"
 #endif
+#define LUV_SOURCE
 #include "luv.h"
+
+#if defined(__aarch64__)
+#define lua_pushlightuserdata(L, P)  lua_pushinteger((L), (lua_Integer)(P))
```

Comment:
Is this a complete enough fix? `lua_islightuserdata` will not work, for example. Maybe we should just avoid using light userdata instead?

The only places its used right now are:
- https://github.com/luvit/luv/blob/9e6e4f2f98647005d5dc0154ca9161540e73f499/src/thread.c#L118-L129
- https://github.com/luvit/luv/blob/9e6e4f2f98647005d5dc0154ca9161540e73f499/src/work.c#L174-L177
- https://github.com/luvit/luv/blob/9e6e4f2f98647005d5dc0154ca9161540e73f499/src/work.c#L260-L263

Reply:
I think that's enough, because  we use use lua_pushlightuserdata just as a key for registry table, and not use it as a real pointer or userdata,  lua_pushlightuserdata is keep in luv private source code. As for lua_islightuserdata, we never use it.
The best solution is to remove lightuserdata api from luv, but that large change,  so I use define to get same result.

Reply:
wait here https://github.com/luvit/luv/blob/9e6e4f2f98647005d5dc0154ca9161540e73f499/src/thread.c#L72-L74 maybe wrong.